src/social_auth/backends/visualstudio.py

```diff
@@ -0,0 +1,79 @@
+"""
+Obtain
+VISUALSTUDIO_CLIENT_ID & VISUALSTUDIO_CLIENT_SECRET
+and put into sentry.conf.py
+"""
+from __future__ import absolute_import
+
+import requests
+
+from social_auth.backends import BaseOAuth2, OAuthBackend
+
+VISUALSTUDIO_AUTHORIZATION_URL = 'https://app.vssps.visualstudio.com/oauth2/authorize'
+VISUALSTUDIO_TOKEN_EXCHANGE_URL = 'https://app.vssps.visualstudio.com/oauth2/token'
+VISUALSTUDIO_USER_DETAILS_URL = 'https://app.vssps.visualstudio.com/_apis/profile/profiles/me?api-version=1.0'
+
+
+class VisualStudioBackend(OAuthBackend):
+    """Visual Studio OAuth authentication backend"""
+    name = 'visualstudio'
+    EXTRA_DATA = [
+        ('id', 'id'),
+        ('refresh_token', 'refresh_token'),
+    ]
+
+    def get_user_details(self, response):
+        """Return user details from Visual Studio account"""
+        return {
+            'email': response.get('email'),
+            'id': response.get('id'),
+            'full_name': response.get('full_name')
+        }
+
+
+class VisualStudioAuth(BaseOAuth2):
+    """Slack OAuth authentication mechanism"""
+    AUTHORIZATION_URL = VISUALSTUDIO_AUTHORIZATION_URL
+    ACCESS_TOKEN_URL = VISUALSTUDIO_TOKEN_EXCHANGE_URL
+    AUTH_BACKEND = VisualStudioBackend
+    SETTINGS_KEY_NAME = 'VISUALSTUDIO_APP_ID'
+    SETTINGS_SECRET_NAME = 'VISUALSTUDIO_CLIENT_SECRET'
```

Comment:
we're kind of abusing this keyname here, as it has both a "CLIENT SECRET" and an "APP SECRET" (which is the historical use of client_secret)